src/java/arjdbc/jdbc/RubyJdbcConnection.java

```diff
@@ -2693,51 +3267,46 @@ protected RubyArray mapColumnsResult(final ThreadContext context,
         final DatabaseMetaData metaData, final TableName components, final ResultSet results,
-        final RubyClass Column, final boolean lookupCastType, final boolean setPrimary)
+        final RubyClass Column)
         throws SQLException {
 
-        final Ruby runtime = context.getRuntime();
-
-        final Collection<String> primaryKeyNames =
-            setPrimary ? getPrimaryKeyNames(metaData, components) : null;
+        final Ruby runtime = context.runtime;
 
-        final RubyArray columns = runtime.newArray();
-        final IRubyObject config = getConfig(context);
+        final RubyArray columns = RubyArray.newArray(runtime);
         while ( results.next() ) {
             final String colName = results.getString(COLUMN_NAME);
-            final RubyString railsColumnName = RubyString.newInternalFromJavaExternal(runtime, caseConvertIdentifierForRails(metaData, colName));
+            final RubyString columnName = cachedString(context, caseConvertIdentifierForRails(metaData, colName));
             final IRubyObject defaultValue = defaultValueFromResultSet( runtime, results );
-            final RubyString sqlType = RubyString.newInternalFromJavaExternal( runtime, typeFromResultSet(results) );
+            final RubyString sqlType = cachedString(context, typeFromResultSet(results));
             final RubyBoolean nullable = runtime.newBoolean( ! results.getString(IS_NULLABLE).trim().equals("NO") );
-            final IRubyObject[] args;
-            if ( lookupCastType ) {
-                final IRubyObject castType = getAdapter(context).callMethod(context, "lookup_cast_type", sqlType);
-                args = new IRubyObject[] {config, railsColumnName, defaultValue, castType, sqlType, nullable};
-            } else {
-                args = new IRubyObject[] {config, railsColumnName, defaultValue, sqlType, nullable};
-            }
 
-            IRubyObject column = Column.callMethod(context, "new", args);
-            columns.append(column);
+            final String tabName = results.getString(TABLE_NAME);
+            final RubyString tableName = cachedString(context, caseConvertIdentifierForRails(metaData, tabName));
 
-            if ( primaryKeyNames != null ) {
-                final RubyBoolean primary = runtime.newBoolean( primaryKeyNames.contains(colName) );
-                column.getInstanceVariables().setInstanceVariable("@primary", primary);
-            }
+            final IRubyObject type_metadata = getAdapter().callMethod(context, "fetch_type_metadata", sqlType);
+
+            // (name, default, sql_type_metadata = nil, null = true, table_name = nil, default_function = nil, collation = nil, comment: nil)
+            final IRubyObject[] args = new IRubyObject[] {
+                columnName, defaultValue, type_metadata, nullable, tableName
+            };
+            columns.append( Column.newInstance(context, args, Block.NULL_BLOCK) );
         }
         return columns;
     }
 
+    @Deprecated
```

Comment:
I might have missed others but I would like to just kill deprecated methods and start fresh.